antora/docs/modules/ROOT/pages/trusted_tasks.adoc

```diff
@@ -66,3 +71,44 @@ compatibility. Will be removed entirely in future.
 <5> The _expires on_ time
 <6> Unique reference of the Task within the provenance URI (git commit id or
 image digest)
+
+== Trusted Artifacts (Current)
+
+Trusted Artifacts is the current approach to securely share files between Tekton tasks in Konflux, especially for custom task use without compromising build integrity.
+
+This approach allows users to include their own Tekton tasks (e.g. unit tests) in a secure and verified way.
+
+Inspired by TEP-0139, this implementation acts as a stop-gap until full upstream support is available.
+
+=== How It Works
+
+* A Task that creates a Trusted Artifact wraps files into an archive.
+
+* The archive location and checksums are exposed via task results.
+
+* A Task that consumes the artifact verifies the checksum and unpacks it into a secure emptyDir volume.
+
+Artifacts are shared via OCI registries (e.g. quay.io), not via Persistent Volume Claims (PVCs).
+
+=== Naming Conventions and Rules
+
+* Producing tasks must use result names ending in _ARTIFACT (e.g. SOURCE_ARTIFACT).
+
+* Consuming tasks must use parameter names ending in _ARTIFACT.
+
+* These tasks must NOT use general-purpose workspaces for file sharing.
+
+* Workspaces are allowed for non-sharing purposes, like mounting Secrets.
+
+=== Migration and Task Variants To aid transition:
+
+* New variants of standard tasks exist, using the suffix -oci-ta, e.g. git-clone-oci-ta.
+
+* New tasks with fresh functionality that use Trusted Artifacts need not follow the suffix naming convention.
+
+
```

Comment:
This is pretty good. I think there's a little more to explain, something like this:

```
When using Trusted Artifacts, only the tasks that directly impact the pipeline's output must be in the trusted task list. Typically this is the git clone task, the prefetch dependencies task, and the build tasks.

In contrast, when using the non-trusted artifacts pipeline, **every** task in the pipeline must be in the trusted task list.
```

I don't think this is perfect explanation, but that's the main thing we want people to understand.

Reply:
What if I'm using the "Trusted Artifacts" (TA) version of the build pipeline and have included a custom task as well?

Now Conforma would be okay with my custom task because the build-influencing tasks (clone, prefetch, and build) use TA. But is it still possible to include a malicious build-influencing logic in my custom task that could compromise the security of my build? If not, why not?

Reply:
The idea is that the malicious task could not influence the output of the trusted build task, because the build task is only accessing data from the trusted artifact produced by (one or more) trusted tasks.

I'm not saying this is bulletproof, but it should prevent a range of code tampering attack vectors, such as "git-clone" -> "modify code, add malware" -> "buildah build". The shared workspace used in non-TA pipelines makes that kind of attack quite doable if there are non-trusted, hence potentially malware, tasks allowed in the pipeline.

I guess also we are counting on (or hoping that) Tekton and Kubernetes to provide enough isolation so one task can't reach out and do bad things to another task or pod while it runs.

Reply:
This helps a lot, thanks!